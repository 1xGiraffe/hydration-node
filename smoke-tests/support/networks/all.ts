import { Config } from './types'

import hydraDXConfig from './hydraDX'
import polkadotConfig from './polkadot'

const all = {
  hydraDX: hydraDXConfig,
  polkadot: polkadotConfig,
} satisfies Record<string, Config>

export default all
