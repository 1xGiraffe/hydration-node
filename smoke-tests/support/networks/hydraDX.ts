import { Config } from './types'

export default {
  polkadot: {
    name: 'hydraDX' as const,
    endpoint: 'wss://rpc.hydradx.cloud',
  },
  kusama: {
    name: 'basilisk' as const,
    endpoint: 'wss://rpc.basilisk.cloud',
  },
  config: ({ alice }) => ({
    storages: {
      System: {
        Account: [[[alice.address], {data: {free: "1000000000000000000"}}]],
      },
      Tokens: {
        accounts: [
          [[alice.address, 2], {free: '0x0000000000000000000000000000000000000000000000056bc75e2d6310000'}],
          [[alice.address, 5], {free: 100e10}],
        ],
      },
      Council: {
        members: [
          alice.address,
        ]
      },
      TechnicalCommittee: {
        members: [
          alice.address,
        ]
      },


    },
  }),
} satisfies Config

export const hydraDX = {
  paraId: 2034,
  dai: 2,
  hdx: 0,
  dot: 5,
}

export const basilisk = {
  paraId: 2090,
  dai: 13,
}
