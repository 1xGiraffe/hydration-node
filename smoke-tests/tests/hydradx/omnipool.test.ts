import { afterAll, assert, beforeEach, describe, it } from 'vitest'
import { sendTransaction, testingPairs } from '@acala-network/chopsticks-testing'

import { checkEvents } from '../../support/helpers'
import { createNetworks } from '../../support/networks'

import { hydraDX } from '../../support/networks/hydraDX'
import { query } from '../../support/helpers/api'
import {queryTokenBalance} from "../../support/helpers/api/query";

const s = "0x0000000000000000000000000000000000000000000000056bc75e2d63100000";

describe.each([
  {
    name: 'hydraDX',
    asset: [hydraDX.dot,  2e10],
  },
  {
    name: 'hydraDX',
    asset: [hydraDX.dai,  "0x0000000000000000000000000000000000000000000000056bc75e2d63100000"],
  }
] as const)('$name omnipool liquidity', async ({ name, asset}) => {
  const { [name]: chain } = await createNetworks({ [name]: undefined })
  const { alice } = testingPairs()

  const head = chain.chain.head

  afterAll(async () => {
    await chain.teardown()
  })

  beforeEach(async () => {
    await chain.chain.setHead(head)
  })

  it.each([
    {
      name: 'addLiquidity',
      tx: chain.api.tx.omnipool.addLiquidity(asset[0], asset[1]),
    }
  ])('$name works', async ({ tx }) => {
    const tx0 = await sendTransaction(tx.signAsync(alice))
    await chain.chain.newBlock()
    await checkEvents(tx0, 'omnipool', 'tokens').redact({ number: true }).toMatchSnapshot()
  })
})

describe.each([
  {
    name: 'hydraDX',
    trade: [hydraDX.dot, hydraDX.dai, 1e10],
  }
] as const)('$name omnipool trades', async ({ name, trade }) => {
  const { [name]: chain } = await createNetworks({ [name]: undefined })
  const { alice } = testingPairs()

  const head = chain.chain.head

  afterAll(async () => {
    await chain.teardown()
  })

  beforeEach(async () => {
    await chain.chain.setHead(head)
  })

  it.each([
    {
      name: 'sell',
      tx: chain.api.tx.omnipool.sell(trade[0], trade[1], trade[2], 0),
    }
  ])('$name works', async ({ tx }) => {
    const tx0 = await sendTransaction(tx.signAsync(alice))

    await chain.chain.newBlock()
    const lp: any = await query.tokens(trade[1])(chain, alice.address)
    assert(lp.free > 0)
    await checkEvents(tx0, 'omnipool', 'tokens').redact({ number: true }).toMatchSnapshot()
  })
})

