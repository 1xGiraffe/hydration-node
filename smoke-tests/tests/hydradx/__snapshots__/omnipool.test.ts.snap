// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`'hydraDX' omnipool liquidity > 'addLiquidity' works > events 1`] = `[]`;

exports[`'hydraDX' omnipool liquidity > 'addLiquidity' works > events 2`] = `[]`;

exports[`'hydraDX' omnipool trades > 'sell' works > events 1`] = `
[
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "from": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
      "to": "7L53bUTBbfuj14UpdCNPwmgzzHSsrsTWBHX5pys32mVWM3C1",
    },
    "method": "Transfer",
    "section": "tokens",
  },
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "from": "7L53bUTBbfuj14UpdCNPwmgzzHSsrsTWBHX5pys32mVWM3C1",
      "to": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "Transfer",
    "section": "tokens",
  },
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "who": "7L53bUTBbfuj14UpdCNPwmgzzHSsrsTWBHX5pys32mVWM3C1",
    },
    "method": "Withdrawn",
    "section": "tokens",
  },
  {
    "data": {
      "amountIn": "(number)",
      "amountOut": "(number)",
      "assetIn": "(number)",
      "assetOut": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "SellExecuted",
    "section": "omnipool",
  },
]
`;
